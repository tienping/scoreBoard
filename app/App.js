/**
 *
 * App - Main Entry Point
 *
 */
import 'babel-polyfill';
import { Provider } from 'react-redux';
import configureStore from 'hermo/configureStore';
import { Text } from 'react-native';

// import FirebaseRemoteConfig from 'utils/remoteConfigService';
// import pushAttendanceNotification from 'utils/notificationService';

import { Navigation } from 'react-native-navigation';
import { registerScreens } from 'hermo/HermornScreens';

export const store = configureStore({});
registerScreens(store, Provider);

Navigation.startSingleScreenApp({
    screen: {
        screen: 'hermorn.screen.Splash',
        title: 'HERMO',
        navigatorStyle: {
            navBarHidden: true,
            tabBarHidden: true,
        },
    },
});

// console.disableYellowBox = true;

// pushAttendanceNotification();
// FirebaseRemoteConfig();

Text.defaultProps.allowFontScaling = false;
export default function App() {

}
