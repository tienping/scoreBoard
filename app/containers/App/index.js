/**
 *
 * App
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 */

import React from 'react';
import { Helmet } from 'react-helmet';
import styled from 'styled-components';
import { Switch, Route } from 'react-router-dom';

import HomePage from 'containers/HomePage/Loadable';
import EditPage from 'containers/EditPage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import Header from 'components/Header';
import Footer from 'components/Footer';

const AppWrapper = styled.div`
  max-width: calc(768px + 16px * 2);
  margin: 0 auto;
  display: flex;
  min-height: 100%;
  padding: 0 16px;
  flex-direction: column;
`;

export default function App() {
  return (
    <AppWrapper>
      <Helmet titleTemplate="%s - ScoreBoard" defaultTitle="ScoreBoard">
        <meta name="description" content="A React.js Boilerplate application" />
      </Helmet>
      <Header />
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route path="/edit" component={EditPage} />
        <Route path="" component={NotFoundPage} />
      </Switch>
      <Footer />
    </AppWrapper>
  );
}
