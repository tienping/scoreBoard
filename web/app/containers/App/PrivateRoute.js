import React from 'react';
import PropTypes from 'prop-types';
import { Route } from 'react-router-dom';
// import LoginForm from 'containers/LoginForm/Loadable';

const PrivateRoute = ({ component: Component, auth, ...remainingProps }) => (
    <Route
        render={(props) => (
            // auth
            // ? (<Component {...props} />)
            // : (<LoginForm />)
            <Component {...props}></Component>
        )}
        {...remainingProps}
    />
);

PrivateRoute.propTypes = {
    component: PropTypes.any,
    auth: PropTypes.bool.isRequired,
};

export default PrivateRoute;
