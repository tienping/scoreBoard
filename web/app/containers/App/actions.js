/*
 *
 * App actions
 *
 */

import {
    FETCH_CONFIG,
    FETCH_CONFIG_SUCCESS,
    FETCH_CONFIG_FAILED,
} from './constants';

export function fetchConfig() {
    return {
        type: FETCH_CONFIG,
    };
}

export function fetchConfigSuccess(response) {
    return {
        type: FETCH_CONFIG_SUCCESS,
        payload: response,
    };
}

export function fetchConfigFailed(response) {
    return {
        type: FETCH_CONFIG_FAILED,
        payload: response,
    };
}
