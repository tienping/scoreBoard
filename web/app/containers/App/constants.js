/*
 *
 * App constants
 *
 */
export const DEFAULT_LOCALE = 'en';

export const FETCH_CONFIG = 'app/App/FETCH_CONFIG';
export const FETCH_CONFIG_SUCCESS = 'app/App/FETCH_CONFIG_SUCCESS';
export const FETCH_CONFIG_FAILED = 'app/App/FETCH_CONFIG_FAILED';
