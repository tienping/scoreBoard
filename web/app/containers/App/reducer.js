/*
 *
 * App reducer
 *
 */

import { fromJS } from 'immutable';

import {
    FETCH_CONFIG,
    FETCH_CONFIG_SUCCESS,
    FETCH_CONFIG_FAILED,
} from './constants';

const initialState = fromJS({
    config: {},
});

function appReducer(state = initialState, action) {
    switch (action.type) {
        case FETCH_CONFIG:
            return state;
        case FETCH_CONFIG_SUCCESS:
            return state
                .set('config', action.payload);
        case FETCH_CONFIG_FAILED:
            return state
                .set('config', action.payload);
        default:
            return state;
    }
}

export default appReducer;
