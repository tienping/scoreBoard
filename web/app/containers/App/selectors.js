import { createSelector } from 'reselect';

const selectGlobal = (state) => state.get('global');
const selectRoute = (state) => state.get('route');
const selectSession = (state) => state.get('session');

const makeSelectAuthenticated = () => createSelector(
    selectSession,
    (substate) => substate.get('authenticated')
);

const makeSelectChecked = () => createSelector(
    selectSession,
    (substate) => substate.get('checked')
);

const makeSelectUserData = () => createSelector(
    selectSession,
    (substate) => substate.get('user')
);

const makeSelectLocation = () => createSelector(
    selectRoute,
    (routeState) => routeState.get('location')
);

export {
    selectGlobal,
    selectRoute,
    selectSession,
    makeSelectAuthenticated,
    makeSelectChecked,
    makeSelectUserData,
    makeSelectLocation,
};
