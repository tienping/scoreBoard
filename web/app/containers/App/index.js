/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import { Switch, Route } from 'react-router-dom';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';

import Notify from 'containers/Notify';

import HomePage from 'containers/HomePage/Loadable';
import LoginForm from 'containers/LoginForm/Loadable';
import MallPage from 'containers/MallPage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';

import Topbar from 'containers/Topbar';
import Sidebar from 'containers/Sidebar';
import theme from '../../theme';

import {
    makeSelectAuthenticated,
    makeSelectLocation,
} from './selectors';
import reducer from './reducer';
import saga from './saga';

import {
    fetchConfig,
} from './actions';

import PrivateRoute from './PrivateRoute';

const sidebarWidth = '48px';
const topbarHeight = '40px';

export const HershopTopbar = styled.div`
    top: 0;
    left: 0;
    right: 0;
    z-index: 2000;
    position: absolute;
    height: ${topbarHeight};
    background-color: ${(props) => props.theme.main_bg};
`;
const HershopSideBar = styled.div`
    top: 0;
    left: 0;
    bottom: 0;
    z-index: 1000;
    margin-top: 20px;
    position: absolute;
    background-color: white;
    width: ${sidebarWidth};
    padding-top: ${topbarHeight};
    box-shadow: 0 0 2px rgba(0, 0, 0, 0.3);
`;

const HershopContent = styled.div`
    padding: 16px;
    margin-top: ${topbarHeight};
    margin-left: ${sidebarWidth};
`;

const HershopMobileBar = styled.div`
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 2000;
    overflow: hidden;
    position: absolute;
    background-color: papayawhip;
`;

export class App extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
    componentDidMount() {
        this.props.dispatch(fetchConfig());
    }

    render() {
        const { authenticated } = this.props;
        return (
            <ThemeProvider theme={theme}>
                <section>
                    {authenticated &&
                        <HershopTopbar
                            id="hershop-topbar-container"
                            className="d-none d-md-block"
                        >
                            <Topbar />
                        </HershopTopbar>
                    }
                    <Notify></Notify>
                    {authenticated &&
                        <HershopSideBar
                            id="hershop-sidebar-container"
                            className="d-none d-md-block"
                        >
                            <Sidebar />
                        </HershopSideBar>
                    }
                    <HershopContent id="hershop-content-container">
                        <Switch>
                            <PrivateRoute exact path="/" auth={authenticated} component={HomePage} />
                            <Route exact path="/mall" component={MallPage} />
                            <Route exact path="/mall/:id" component={MallPage} />
                            {/* <Route exact path="/flagship" auth={authenticated} component={Flagship} />
                            <Route exact path="/flagship/:id" auth={authenticated} component={Flagship} /> */}

                            <Route exact path="/login" component={LoginForm} />
                            <Route path="" component={NotFoundPage} />
                        </Switch>
                    </HershopContent>
                    <HershopMobileBar
                        id="hershop-mobilebar-container"
                        className="d-block d-md-none"
                    >
                        This is bottom mobile bar
                    </HershopMobileBar>
                </section>
            </ThemeProvider>
        );
    }
}

App.propTypes = {
    dispatch: PropTypes.func.isRequired,
    authenticated: PropTypes.bool.isRequired,
    // location: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
    authenticated: makeSelectAuthenticated(),
    location: makeSelectLocation(),
});

function mapDispatchToProps(dispatch) {
    return {
        dispatch,
    };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);

const withReducer = injectReducer({ key: 'config', reducer });
const withSaga = injectSaga({ key: 'config', saga });

export default compose(
    withReducer,
    withSaga,
    withConnect,
)(App);
