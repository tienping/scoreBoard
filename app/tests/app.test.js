import React from 'react';
import { Root } from 'native-base';
import App from '../App';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() });

describe('App', () => {
    it('should render correctly', () => {
        const tree = shallow(<App />)
        expect(tree).toMatchSnapshot();
    });
});
