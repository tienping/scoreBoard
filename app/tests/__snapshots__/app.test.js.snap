// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Root)
        virtual={undefined}
      >
        <withReducer(withSaga(Connect(Navigation)))>
          <withReducer(Connect(App)) />
        </withReducer(withSaga(Connect(Navigation)))>
      </Styled(Root)>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "injectedReducers": Object {},
        "injectedSagas": Object {},
        "replaceReducer": [Function],
        "runSaga": [Function],
        "subscribe": [Function],
        Symbol(Symbol.observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <withReducer(withSaga(Connect(Navigation)))>
          <withReducer(Connect(App)) />
        </withReducer(withSaga(Connect(Navigation)))>,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <withReducer(Connect(App)) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Root)
          virtual={undefined}
        >
          <withReducer(withSaga(Connect(Navigation)))>
            <withReducer(Connect(App)) />
          </withReducer(withSaga(Connect(Navigation)))>
        </Styled(Root)>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "injectedReducers": Object {},
          "injectedSagas": Object {},
          "replaceReducer": [Function],
          "runSaga": [Function],
          "subscribe": [Function],
          Symbol(Symbol.observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <withReducer(withSaga(Connect(Navigation)))>
            <withReducer(Connect(App)) />
          </withReducer(withSaga(Connect(Navigation)))>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <withReducer(Connect(App)) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
